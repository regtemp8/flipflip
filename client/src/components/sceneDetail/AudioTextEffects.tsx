import { Card, CardContent, Grid } from '@mui/material'

import AudioCard from '../configGroups/AudioCard'
import TextCard from '../configGroups/TextCard'

export interface AudioTextEffectsProps {
  sceneID: number
}

export default function AudioTextEffects(props: AudioTextEffectsProps) {
  return (
    <Grid container spacing={2}>
      <Grid item xs={12} lg={6}>
        <Card>
          <CardContent>
            <AudioCard sceneID={props.sceneID} startPlaying={false} />
          </CardContent>
        </Card>
      </Grid>
      <Grid item xs={12} md={6}>
        <Card>
          <CardContent>
            <TextCard sceneID={props.sceneID} />
          </CardContent>
        </Card>
      </Grid>
    </Grid>
  )
}

;(AudioTextEffects as any).displayName = 'AudioTextEffects'
