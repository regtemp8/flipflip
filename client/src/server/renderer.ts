import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
