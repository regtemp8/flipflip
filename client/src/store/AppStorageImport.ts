import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { EntryState } from "./EntryState";
import type Tag from "./tag/Tag";
import type SceneGroup from "./sceneGroup/SceneGroup";
import type SceneGrid from "./sceneGrid/SceneGrid";
import type Scene from "./scene/Scene";
import type Playlist from "./playlist/Playlist";
import type Overlay from "./overlay/Overlay";
import type LibrarySource from "./librarySource/LibrarySource";
import type Clip from "./clip/Clip";
import type CaptionScript from "./captionScript/CaptionScript";
import type Audio from "./audio/Audio";
import type App from "./app/data/App";

export interface AppStorageImport {
  app: App;
  audio: EntryState<Audio>;
  captionScript: EntryState<CaptionScript>;
  clip: EntryState<Clip>;
  librarySource: EntryState<LibrarySource>;
  overlay: EntryState<Overlay>;
  playlist: EntryState<Playlist>;
  scene: EntryState<Scene>;
  sceneGrid: EntryState<SceneGrid>;
  sceneGroup: EntryState<SceneGroup>;
  tag: EntryState<Tag>;
}

export const initialAppStorageImport: AppStorageImport = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
}