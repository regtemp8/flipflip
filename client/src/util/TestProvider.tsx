import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
