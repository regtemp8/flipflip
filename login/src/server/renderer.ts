import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
