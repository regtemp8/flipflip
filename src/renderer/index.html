<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
