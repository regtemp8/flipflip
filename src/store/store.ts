import { configureStore } from '@reduxjs/toolkit'
import appSlice from './app/slice'
import audioSlice from './audio/slice'
import constantsSlice from './constants/slice'
import playersSlice from './player/slice'
import captionScriptSlice from './captionScript/slice'
import clipSlice from './clip/slice'
import librarySourceSlice from './librarySource/slice'
import overlaySlice from './overlay/slice'
import playlistSlice from './playlist/slice'
import sceneSlice from './scene/slice'
import sceneGridSlice from './sceneGrid/slice'
import sceneGroupSlice from './sceneGroup/slice'
import tagSlice from './tag/slice'
import videoClipperSlice from './videoClipper/slice'
import scenePickerSlice from './scenePicker/slice'
import captionScriptorSlice from './captionScriptor/slice'
import sceneDetailSlice from './sceneDetail/slice'
import audioLibrarySlice from './audioLibrary/slice'
import sourceScraperSlice from './sourceScraper/slice'

const store = configureStore({
  reducer: {
    app: appSlice,
    audio: audioSlice,
    captionScript: captionScriptSlice,
    clip: clipSlice,
    constants: constantsSlice,
    librarySource: librarySourceSlice,
    overlay: overlaySlice,
    playlist: playlistSlice,
    scene: sceneSlice,
    sceneGrid: sceneGridSlice,
    sceneGroup: sceneGroupSlice,
    tag: tagSlice,
    // components
    players: playersSlice,
    videoClipper: videoClipperSlice,
    scenePicker: scenePickerSlice,
    captionScriptor: captionScriptorSlice,
    sceneDetail: sceneDetailSlice,
    audioLibrary: audioLibrarySlice,
    sourceScraper: sourceScraperSlice
  }
})

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
