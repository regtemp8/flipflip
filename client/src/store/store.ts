/// <reference path="../global.d.ts"/>
import { createReduxStore } from '../server/renderer'

const store = createReduxStore(
  window.flipflipAppStorage,
  window.flipflipConstants
)

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
