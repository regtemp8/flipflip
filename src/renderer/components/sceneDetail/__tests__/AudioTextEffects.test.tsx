import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import AudioTextEffects from "../AudioTextEffects";
import TestProvider from "../../../../../test/util/TestProvider";
import store from "../../../../store/store";

jest.mock('../../configGroups/AudioCard', () => 'AudioCard');
jest.mock('../../configGroups/TextCard', () => 'TextCard');

// TODO create functional tests instead of snapshots
describe("AudioTextEffects", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <AudioTextEffects sceneID={null} />
      </TestProvider>
    );

    let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
