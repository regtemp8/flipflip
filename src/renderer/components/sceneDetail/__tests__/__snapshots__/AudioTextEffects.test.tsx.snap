// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioTextEffects should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6 css-f0xsrb-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <AudioCard
          onAddTracks={[Function]}
          onPlay={[Function]}
          onUpdateScene={[Function]}
          scene={null}
          startPlaying={false}
          systemMessage={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <TextCard
          onAddScript={[Function]}
          onPlay={[Function]}
          onUpdateScene={[Function]}
          scene={null}
          systemMessage={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
